src/main/java/com/google/api/generator/engine/writer/JavaWriterVisitor.java

```diff
@@ -351,6 +358,39 @@ public void visit(TryCatchStatement tryCatchStatement) {
     newline();
   }
 
+  /** =============================== COMMENT =============================== */
+  public void visit(LineComment lineComment) {
+    // Split comments by new line and add `//` to each line.
+    String formattedSource =
+        JavaFormatter.format(
+            String.format("// %s", String.join("\n//", lineComment.comment().split("\\r?\\n"))));
+    buffer.append(formattedSource);
+  }
+
+  public void visit(BlockComment blockComment) {
+    // Split comments by new line and embrace the comment block with `/** */`.
+    String sourceComment = blockComment.comment();
+    String formattedSource =
+        JavaFormatter.format(
+            String.format("%s %s %s", BLOCK_COMMENT_START, sourceComment, BLOCK_COMMENT_END));
+    // TODO(xiaozhenliu): call comment escaper here.
```

Comment:
The comment escaper should be part of BlockComment (as discussed in the last review), could we please update this TODO?

Reply:
updated the comment in BlockComment class, so the comment() in BlockComment class would be escaped, and the formatter will still be called in JavaWriterVisitor.